core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -86,22 +86,23 @@ oppia.factory('responsesService', [
             .can_have_solution && stateSolutionService.savedMemento !== null) {
           if (stateSolutionService.savedMemento.correctAnswer !== null) {
             var currentStateName = editorContextService.getActiveStateName();
-            SolutionHelperService.verifySolution(
+            SolutionVerificationService.verifySolution(
               explorationContextService.getExplorationId(),
               explorationStatesService.getState(currentStateName),
               stateSolutionService.savedMemento.correctAnswer,
               function() {
                 explorationStatesService.updateSolutionValidity(
-                  currentStateName, false);
+                  currentStateName, true);
                 alertsService.addInfoMessage(
-                  'That solution does not lead to the next state!');
+                  'The solution is now valid!');
```

Comment:
Can't you just check SolutionValidityService at the outset for the current status of that state's solution? If it's currently invalid and then becomes valid then you can show the message.

Definitely don't do the second one. It'll irritate the creator :)